### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by jozefbakus**
Renamed all "Trigger" mentions to "Sensor" (Consistency).
Foreign keys fix.

### CODE REVIEW

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js [hunk 1]
```diff
@@ -76,17 +76,17 @@ sap.ui.define([
 		getWorkflowToSave: function () {
 			let workflow = this._model.getProperty("/workflow");
 			let matchProperties = {};
-			workflow.trigger.triggerProperties.matchProperties.map(function(joinCondition) {
+			workflow.sensor.sensorProperties.matchProperties.map(function(joinCondition) {
```
Comment (Zejnilovic) [inline src/main/resources/ui/controller/UpsertWorkflow.controller.js]: This is a bit weird naming. If you know it is a sensor, why call it's properties sensor properties? I would just call them properties.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/Application.scala [hunk 2]
```diff
@@ -6,9 +6,9 @@ import com.fasterxml.jackson.databind._
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
 import com.fasterxml.jackson.module.scala.DefaultScalaModule
 import za.co.absa.hyperdrive.trigger.api.rest.services._
-import za.co.absa.hyperdrive.trigger.models.enums.EventTypes.EventType
+import za.co.absa.hyperdrive.trigger.models.enums.SensorTypes.SensorType
```
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/Application.scala:9]: Same naming logic as a first comment

File: src/main/scala/za/co/absa/hyperdrive/trigger/models/enums/SensorTypes.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  sealed abstract class SensorType(val name: String) {
+    override def toString: String = name
+  }
+
+  case object Kafka extends SensorType("Kafka")
+  case object AbsaKafka extends SensorType("Absa-Kafka")
+
+  val sensorTypes: Set[SensorType] = Set(Kafka, AbsaKafka)
+
+}
```
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/hyperdrive/trigger/models/enums/SensorTypes.scala]: missing new line at the end of a file

File: src/main/scala/za/co/absa/hyperdrive/trigger/models/tables/SensorTable.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        sensor.workflowId,
+        sensor.sensorType,
+        sensor.sensorProperties.properties.variables,
+        sensor.sensorProperties.properties.maps,
+        sensor.sensorProperties.matchProperties,
+        sensor.id
+      )
+  )
+
+}
```
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/hyperdrive/trigger/models/tables/SensorTable.scala]: Missing new line at the end of the file

**Review by Zejnilovic — COMMENTED** [review]
You are missing copyright for new files

**Comment by jozefbakus** [comment]
> You are missing copyright for new files

Yes, I am aware of that. It will be fixed in separate PR (You have raised bug for that.)

File: src/main/scala/za/co/absa/hyperdrive/trigger/models/tables/SensorTable.scala (hunk 4 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/models/tables/SensorTable.scala]: Fixed.

File: src/main/scala/za/co/absa/hyperdrive/trigger/models/enums/SensorTypes.scala (hunk 3 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/models/enums/SensorTypes.scala]: Fixed.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/Application.scala (hunk 2 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/Application.scala:9]: Fixed (nope) ... Hmm, this one is something else. I wouldn't change it. SensorTypes is "enum" and SensorType is passed around without context so it is more clear when it is named this way. For example: 
 `  implicit val sensorTypeMapper: JdbcType[SensorType] =
    MappedColumnType.base[SensorType, String](
      sensorType => sensorType.name,
      sensorTypeName => SensorTypes.sensorTypes.find(_.name == sensorTypeName).getOrElse(
        throw new Exception(s"Couldn't find SensorType: $sensorTypeName")
      )
    )
`
JdbcType[Type] wouldn't make sense.

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js (hunk 1 above)
Comment (jozefbakus) [reply src/main/resources/ui/controller/UpsertWorkflow.controller.js]: It has been fixed as discussed.

**Review by Zejnilovic — APPROVED** [review]
Looks cool

**Pull request merged**